Mathlib/Geometry/Euclidean/Angle/Unoriented/Basic.lean

```diff
@@ -328,4 +334,26 @@ theorem sin_eq_one_iff_angle_eq_pi_div_two : sin (angle x y) = 1 ↔ angle x y =
   rw [← cos_eq_zero_iff_angle_eq_pi_div_two, ← abs_eq_zero, abs_cos_eq_sqrt_one_sub_sin_sq, h]
   simp
 
-end InnerProductGeometry
+open Matrix
+
+/-- Returns the l2 norm of a vector. This function is necessary to make sure the l2 norm is taken
+for members of `Fin 3 → ℝ` instead of the sup norm-/
+noncomputable def l2Norm (v : EuclideanSpace ℝ (Fin 3)) : ℝ :=
+  ‖v‖
+
+/- The norm of the cross product of two real vectors equals the product of their individual norms
+  times the sine of the angle between them. -/
+theorem crossProduct_norm_eq_norm_mul_norm_mul_sin (a b : EuclideanSpace ℝ (Fin 3)) :
+    l2Norm (a ×₃ b) = ‖a‖ * ‖b‖ * sin (angle a b) := by
```

Comment:
The correct spelling here is the ugly `‖(WithLp.equiv _ _).symm (WithLp.equiv _ _ a ×₃ WithLp.equiv _ _ b)‖ = ...`

Reply:
For my own understanding, why is this preferable to my existing implementation of l2Norm, which has the same end result but is shorter?

Reply:
Because `l2Norm` has no lemmas about it, but there are many lemmas about `(WithLp.equiv _ _)`.

Framed another way, `‖(WithLp.equiv _ _).symm (WithLp.equiv _ _ a ×₃ WithLp.equiv _ _ b)‖` is something that could naturally appear over the course of a proof. If `a := !₂[1, 2, 3]`, then `WithLp.equiv _ _ a` simplifies to `![1, 2, 3]`.

Reply:
Gotcha, thanks!